### PR TITLE
Refactor views by using presenters

### PR DESCRIPTION
**Pull request by cveneziani**

### CODE REVIEW

File: apps/web/presenters/housing_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class HousingPresenter
+  include Hanami::Presenter
+
+  def card_partial_name
+    imported ? :housing_card : :housing_card_pending
```
Comment (nfilzi) [inline apps/web/presenters/housing_presenter.rb]: Nice!

File: apps/web/views/trips/show.rb [hunk 2]
```diff
@@ -1,69 +1,36 @@
...
-    def after_js
-      javascript('copy-invitation-link', async: true)
+    def housings
+      locals[:housings].map { |housing| HousingPresenter.new(housing) }
     end
 
-    def days_before_trip
-      (trip.starting_on - Date.today).to_i
+    def trip
+      TripPresenter.new(locals[:trip])
```
Comment (nfilzi) [inline apps/web/views/trips/show.rb]: So, if I get how it works, you are tapping into the exposed variables from the controller action, to wrap them on methods on the view side, that will be available and exposing presenters to the template?

File: apps/web/presenters/trip_presenter.rb [hunk 3]
```diff
@@ -25,4 +28,32 @@ def ongoing?
...
+        else
+          "<span class='icon'>👆</span Let's create <strong>your first housing!</strong>"
+        end
+      when :ongoing
+        "Time to <strong>enjoy</strong> this trip!"
+      when :completed
+        "Hope you <strong>enjoyed</strong> this trip!"
+      end
+
+    raw message
```
Comment (nfilzi) [inline apps/web/presenters/trip_presenter.rb]: coming from `Hanami::Helpers::EscapeHelper`?

File: apps/web/templates/trips/index.html.erb [hunk 4]
```diff
@@ -31,17 +31,17 @@
...
+                  <% if trip.future? %>
+                    in <strong><%= trip.days_before_beginning %> days</strong>
                   <% end %>
                 </small>
               </p>
 
               <ul class="list-unstyled details">
                 <li>
-                  🗓 <%= format_trip_date(trip.starting_on) %>
+                  🗓 <%= trip.date(trip.starting_on) %>
```
Comment (nfilzi) [inline apps/web/templates/trips/index.html.erb]: A bit weird, but I get the reasoning.

File: apps/web/config/routes.rb [hunk 5]
```diff
@@ -11,14 +11,11 @@
 
 resource :profile, only: [:edit, :update]
 
-get '/trips/:id/token/:token', to: 'trips#show', as: :trip_by_token
+# custom trip routes
+get '/trips/:status',          to: 'trips#index', as: :trips_by_status, status: /(ongoing|completed)/
```
Comment (nfilzi) [inline apps/web/config/routes.rb]: 👍

File: apps/web/views/trips/index.rb [hunk 6]
```diff
@@ -1,14 +1,11 @@
...
     include Web::View
-    include Web::Helpers::DateFormatter
-
-    def days_before_trip(trip)
-      (trip.starting_on - Date.today).to_i
-    end
 
-    def format_trip_date(date)
-      format_date(date, format: '%b %e', ordinal_indicator: true)
+    def trips
```
Comment (nfilzi) [inline apps/web/views/trips/index.rb]: So, if I get how it works, you are tapping into the exposed variables from the controller action, to wrap them on methods on the view side, that will be available and exposing presenters to the template?

File: apps/web/presenters/trip_presenter.rb (hunk 3 above)
Comment (cveneziani) [reply apps/web/presenters/trip_presenter.rb]: Exactly

File: apps/web/templates/trips/index.html.erb (hunk 4 above)
Comment (cveneziani) [reply apps/web/templates/trips/index.html.erb]: Well, I wanted to name it `format_date` but... yep, format_date is already the name of the Hanami helper method we're using. Thus I cannot use it.

Comment (cveneziani) [reply apps/web/templates/trips/index.html.erb]: Do you have a better naming? 😬

File: apps/web/views/trips/index.rb (hunk 6 above)
Comment (cveneziani) [reply apps/web/views/trips/index.rb]: yep

File: apps/web/templates/trips/index.html.erb (hunk 4 above)
Comment (nfilzi) [reply apps/web/templates/trips/index.html.erb]: Nope 😅

**Pull request merged**